tests/Integration/Subscriber/CDN/RocketCDN/AdminPageSubscriber/displayRocketcdnStatus.php

```diff
@@ -0,0 +1,168 @@
+		parent::setUp();
+
+		set_current_screen( 'settings_page_wprocket' );
+	}
+
+	public function tearDown() {
+		parent::tearDown();
+
+		delete_transient( 'rocketcdn_status' );
+	}
+
+	private function getActualHtml() {
+		ob_start();
+		do_action( 'rocket_dashboard_after_account_data' );
+
+		return $this->format_the_html( ob_get_clean() );
+	}
+
+	/**
+	 * Test should render the "no subscription" HTML when the subscription status is "cancelled."
+	 */
+	public function testShouldRenderNoSubscriptionHTMLWhenCancelled() {
+		set_transient(
+			'rocketcdn_status',
+			[
+				'is_active'                     => false,
+				'subscription_status'           => 'cancelled',
+				'subscription_next_date_update' => '2020-01-01',
+			],
+			MINUTE_IN_SECONDS
+		);
+
+		$expected = <<<HTML
+<div class="wpr-optionHeader">
+	<h3 class="wpr-title2">Rocket CDN</h3>
+</div>
+<div class="wpr-field wpr-field-account">
+	<div class="wpr-flex wpr-flex--egal">
+		<div>
+			<span class="wpr-title3"></span>
+			<span class="wpr-infoAccount wpr-isInvalid">No Subscription</span>
+		</div>
+		<div>
+			<a href="#page_cdn" class="wpr-button">Get Rocket CDN</a>
+		</div>
+	</div>
+</div>
+HTML;
+
+		$this->assertSame( $this->format_the_html( $expected ), $this->getActualHtml() );
+	}
+
+	/**
+	 * Test should render HTML when the subscription status is "active" but "is_active" is false.
+	 */
+	public function testShouldRenderHTMLWhenIsActiveFalseAndStatusIsActive() {
+		set_transient(
+			'rocketcdn_status',
+			[
+				'is_active'                     => false,
+				'subscription_status'           => 'active',
+				'subscription_next_date_update' => '2020-01-01',
+			],
+			MINUTE_IN_SECONDS
+		);
+
+		$expected = <<<HTML
+<div class="wpr-optionHeader">
+	<h3 class="wpr-title2">Rocket CDN</h3>
+</div>
+<div class="wpr-field wpr-field-account">
+	<div class="wpr-flex wpr-flex--egal">
+		<div>
+			<span class="wpr-title3"></span>
+			<span class="wpr-infoAccount wpr-isInvalid">2020-01-01</span>
+		</div>
+		<div>
+			<a href="#page_cdn" class="wpr-button">Get Rocket CDN</a>
+		</div>		
+	</div>
+</div>
+HTML;
+
+		$this->assertSame( $this->format_the_html( $expected ), $this->getActualHtml() );
+	}
+
+	/**
+	 * Test should render HTML when the subscription status is "cancelled" but "is_active" is true.
+	 */
+	public function testShouldRenderHTMLWhenStatusCancelledAndIsActiveTrue() {
+		set_transient(
+			'rocketcdn_status',
+			[
+				'is_active'                     => true,
+				'subscription_status'           => 'cancelled',
+				'subscription_next_date_update' => '2020-01-01',
+			],
+			MINUTE_IN_SECONDS
+		);
+
+		$expected = <<<HTML
+<div class="wpr-optionHeader">
+	<h3 class="wpr-title2">Rocket CDN</h3>
+</div>
+<div class="wpr-field wpr-field-account">
+	<div class="wpr-flex">
+		<div>
+			<span class="wpr-title3">Next Billing Date</span>
+			<span class="wpr-infoAccount wpr-isValid">No Subscription</span>
+		</div>	
+	</div>
+</div>
+HTML;
```

Comment:
When `'is_active'` is `true` but the `'subscription_status'` is `'cancelled'`, the copy may confuse the consumer. It says "Next Billing Date" and then has "No Subscription".

Reply:
logic was updated to only rely on the `subscription_status`